src/fairseq2/checkpoint.py

```diff
@@ -154,6 +154,16 @@ def get_last_step_number(self, *, with_model: bool = False) -> Optional[int]:
             If ``True``, only considers checkpoints with a model.
         """
 
+    @abstractmethod
+    def get_model_checkpoint_path(
```

Comment:
It would be nice if we could keep the base `CheckpointManager` "file-system agnostic" since in the future we might have other checkpoint manager types that do not use a file system. Would exposing this function only in `FileCheckpointManager` work for your use case?

Reply:
Yes, it would!
I'll remove it from the base class.